crates/ruma-client-api/src/sync/sliding_sync_events.rs

```diff
@@ -0,0 +1,300 @@
+//! [POST /_matrix/client/unstable/org.matrix.msc3575/sync](https://github.com/matrix-org/matrix-doc/blob/kegan/sync-v3/proposals/3575-sync.md)
+
+use std::{collections::BTreeMap, time::Duration};
+
+use js_int::UInt;
+use ruma_common::{
+    api::ruma_api,
+    events::{
+        AnySyncRoomEvent, AnySyncStateEvent, AnyStrippedStateEvent,
+        RoomEventType
+    },
+    OwnedRoomId,
+    serde::Raw, serde::duration::opt_ms,
+};
+use serde::{Deserialize, Serialize};
+use super::sync_events::v3::UnreadNotificationsCount;
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct SyncRequestListFilters {
+    // All fields are applied with AND operators, hence if is_dm:true and is_encrypted:true
+    // then only Encrypted DM rooms will be returned. The absence of fields implies no filter
+    // on that criteria: it does NOT imply 'false'.
+    // These fields may be expanded through use of extensions.
+
+    /// Sticky. Flag which only returns rooms present (or not) in the DM section of account data.
+    /// If unset, both DM rooms and non-DM rooms are returned. If false, only non-DM rooms
+    /// are returned. If true, only DM rooms are returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_dm: Option<bool>,
+
+    /// Sticky. A list of spaces which target rooms must be a part of. For every invited/joined room for
+    /// this user, ensure that there is a parent space event which is in this list. If unset, all
+    /// rooms are included. Servers MUST NOT navigate subspaces. It is up to the client to
+    /// give a complete list of spaces to navigate. Only rooms directly in these spaces will be
+    /// returned.
```

Comment:
Feels a bit excessive and repetitive.

```suggestion
    /// A list of spaces which target rooms must be directly a part of. Servers will not navigate subspaces, so only rooms directly in these spaces will be returned.
    /// 
    /// If unset, all rooms are included.
    ///
    /// Sticky: Yes
```